File: src/main/java/src/gestorAplicacion/Calendario/gestionDatos.java
package gestorAplicacion.Calendario;

import java.io.Serializable;
import java.util.ArrayList;
import java.util.Arrays;
import java.util.List;

import baseDatos.Deserializador;
import gestorAplicacion.personas.Estudiante;
import gestorAplicacion.personas.Profesor;

public class gestionDatos implements Serializable{
	
	private static final long serialVersionUID = 1L;
	
	private Facultad facultadMinas = new Facultad();
	private ArrayList<Estudiante> estudiantes = new ArrayList<Estudiante>();
	private ArrayList<Materia> materias = facultadMinas.getMaterias();
	private ArrayList<Profesor> profesores = facultadMinas.getProfesores();
	private Beca sistemaBecas = new Beca("");
	
	public gestionDatos() {
			Deserializador.deserializar(this);
		}
	
	//getters y setters
		public ArrayList<Materia> getMaterias () {
			return materias;
		}
		public void setMaterias (ArrayList<Materia> materias) {
			this.materias = materias;
		}
		public ArrayList<Estudiante> getEstudiantes () {
			return estudiantes;
		}
		public void setEstudiantes (ArrayList<Estudiante> estudiantes) {
			this.estudiantes = estudiantes;
		}
		public ArrayList<Profesor> getProfesores() {
			return profesores;
		}
		public void setProfesores (ArrayList<Profesor> profesores) {
			this.profesores=profesores;
		}
		public Beca getSistemaBecas () {
			return sistemaBecas;
		}
		public void setSistemaBecas (Beca sistemaBecas) {
			this.sistemaBecas=sistemaBecas;
		}
		
		
		//metodos
		public Estudiante nuevoEstudiante(String nombre, int ID, String Email, boolean fueBecado, ArrayList<Materia> materias_cursadas) {
			Estudiante nuevoEstudiante = new Estudiante(nombre, ID, Email, fueBecado, materias_cursadas);
			estudiantes.add(nuevoEstudiante);
			return nuevoEstudiante;
		}
		public Estudiante nuevoEstudiante(String nombre, int ID, String Email, boolean fueBecado) {
			Estudiante nuevoEstudiante = new Estudiante(nombre, ID, Email, fueBecado);
			estudiantes.add(nuevoEstudiante);
			return nuevoEstudiante;
		}
		 
}

File: src_py/Calendario/gestionDatos.py
import pickle
import os
from Calendario.Facultad import Facultad
from Calendario.Beca import Beca
from personas.Estudiante import Estudiante
from baseDatos.Deserializador import Deserializador


class gestionDatos:
    def __init__(self):
        self.facultadMinas = Facultad()
        self.estudiantes = []
        self.materias = self.facultadMinas.getMaterias()
        self.profesores = self.facultadMinas.getProfesores()
        self.sistemaBecas = Beca("")
        Deserializador.deserializar(self)

    def getMaterias(self):
        return self.materias

    def setMaterias(self, materias):
        self.materias = materias

    def getEstudiantes(self):
        return self.estudiantes

    def setEstudiantes(self, estudiantes):
        self.estudiantes = estudiantes

    def getProfesores(self):
        return self.profesores

    def setProfesores(self, profesores):
        self.profesores = profesores

    def getSistemaBecas(self):
        return self.sistemaBecas

    def setSistemaBecas(self, sistemaBecas):
        self.sistemaBecas = sistemaBecas

    def getDocumentos(self):
        Documentos = []
        for estudiante in self.getEstudiantes():
            Documentos.append(estudiante.getID())
        return Documentos

    def getDocumentosProfesores(self):
        Documentos = []
        for profesor in self.getProfesores():
            Documentos.append(profesor.getID())
        return Documentos

    def nuevoEstudiante(self, nombre, ID, Email, fueBecado, materias_cursadas=None):
        nuevoEstudiante = Estudiante(nombre, ID, Email, fueBecado, materias_cursadas)
        self.estudiantes.append(nuevoEstudiante)
        return nuevoEstudiante

